### PR TITLE
Update common.vm for all affected packages

### PR DESCRIPTION
**Pull request by emtuls**
From https://github.com/mandiant/VM-Packages/pull/959, there was an issue with packages not requiring the most up to date `common.vm` leading to many packages to fail in the `daily.yml` check.

There was also an issue with some older packages not performing `sha256 checks` due to it not being a mandatory requirement when made, which will need to be fully addressed later.

This should fix these issues.

### CODE REVIEW

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 1]
```diff
@@ -289,19 +289,19 @@ function VM-Install-Shortcut{
...
         }
+
         Install-ChocolateyShortcut @shortcutArgs
 
     } else {
         $shortcutArgs = @{
             ShortcutFilePath = $shortcut
             TargetPath       = $executablePath
-            Arguments        = $arguments
-            WorkingDirectory = $executableDir
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:299]: why are we removing this two arguments here?

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 2]
```diff
@@ -369,34 +364,21 @@ function VM-Install-From-Zip {
             url64bit       = $zipUrl_64
             checksum64     = $zipSha256_64
         }
-        Install-ChocolateyZipPackage @packageArgs | Out-Null
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:372]: why are you changing this?

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 3]
```diff
@@ -639,27 +603,7 @@ function VM-Install-With-Installer {
         Install-ChocolateyInstallPackage @packageArgs
         VM-Assert-Path $executablePath
 
-        # if no icon path provided, set the shortcut icon to be the executable's icon. For MSI files, attempt to get executable icon from the installer first.
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:642]: why are you removing this code?

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 4]
```diff
@@ -533,22 +515,6 @@ function VM-Remove-Tool-Shortcut {
     Remove-Item $shortcut -Force -ea 0 | Out-Null
 }
 
-# Delete Desktop shortcuts
-function VM-Remove-Desktop-Shortcut {
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:537]: why are you removing this code?

**Review by Ana06 — COMMENTED** [review]
It looks to me like you are removing code in common.vm by accident

**Comment by emtuls** [comment]
That would be correct. Too many ctrl-z's 😬 

Thanks for catching that!

**Comment by emtuls** [comment]
Recent fail was due to difference in expected return from `VM-Install-From-Zip`, which it shouldn't need to do anyway, so I updated the 2 packages that failed because of that to work in a more intended way.

Those packages were:
`file.vm`
`dnspyex.vm`

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 5]
```diff
@@ -221,7 +221,7 @@ function VM-Install-Raw-GitHub-Repo {
         # $powershellCommand = "Get-Content README.md"
         # $powershellCommand = "Import-Module module.ps1; Get-Help Main-Function"
         [Parameter(Mandatory=$false)]
-        [string] $powershellCommand
+        [string] $powershellCommand = ""
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:224]: why is this needed?

File: packages/dnspyex.vm/tools/chocolateyinstall.ps1 [hunk 6]
```diff
@@ -7,8 +7,9 @@ try {
 
   $zipUrl = "https://github.com/dnSpyEx/dnSpy/releases/download/v6.5.0/dnSpy-netframework.zip"
   $zipSha256 = "5962e3cca902e650c61050e2a879af58b78eec91288b7a7b77a7bc761424a0ed"
-  $toolDir = (VM-Install-From-Zip $toolName $category $zipUrl -zipSha256 $zipSha256)[0]
+  VM-Install-From-Zip $toolName $category $zipUrl -zipSha256 $zipSha256
```
Comment (Ana06) [inline packages/dnspyex.vm/tools/chocolateyinstall.ps1:10]: This function should return the `$toolDir` as documented in this comment in `$common`:
> # This functions returns $toolDir (outputed by Install-ChocolateyZipPackage) and $executablePath

If the previous PR got rid of that, we should bring it back instead of having to recalculate `$toolDir`. Other packages use this as well.

File: packages/file.vm/tools/chocolateyinstall.ps1 [hunk 7]
```diff
@@ -8,8 +8,10 @@ try {
     $zipUrl = "https://github.com/nscaife/file-windows/releases/download/20170108/file-windows-20170108.zip"
     $zipSha256 = "963147318f96d9345471e1a9a3943def4d95fcb3c1fe020e465ab910d0cda4a3"
 
-    $executablePath = (VM-Install-From-Zip $toolName $category $zipUrl -zipSha256 $zipSha256 -consoleApp $true)[-1]
-    $executableDir = Split-Path -Path $executablePath
+    VM-Install-From-Zip $toolName $category $zipUrl -zipSha256 $zipSha256 -consoleApp $true
```
Comment (Ana06) [inline packages/file.vm/tools/chocolateyinstall.ps1:11]: Same here. We should leave it as it was before and fix `VM-Install-From-Zip` to behave as expected.

File: packages/common.vm/tools/vm.common/vm.common.psm1 (hunk 5 above)
Comment (emtuls) [reply packages/common.vm/tools/vm.common/vm.common.psm1:224]: I thought that was what led to: https://github.com/mandiant/VM-Packages/actions/runs/8651261154/job/23721615509#step:4:460

```
[adconnectdump.vm] vm.common.psm1 [+] ERROR : A parameter cannot be found that matches parameter name 'powershellCommand'.
At C:\ProgramData\_VM\vm.common\vm.common.psm1:228 char:185
+ ... utableName $executableName -withoutBinFile -powershellCommand $powers ...
+                                                ~~~~~~~~~~~~~~~~~~
ERROR: A parameter cannot be found that matches parameter name 'powershellCommand'.
The upgrade of adconnectdump.vm was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\adconnectdump.vm\tools\chocolateyinstall.ps1'.
```

But now I'm realizing that it was just due to not having the `common.vm` version set properly in a dependency.


We do the same for `$executableName`: https://github.com/mandiant/VM-Packages/pull/987/files#diff-42c583682dc7cee57079eab399b94fbe5a60b1fcec5626a25a7092e197cb7308L217

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 8]
```diff
@@ -397,7 +399,7 @@ function VM-Install-From-Zip {
             VM-Install-Shortcut -toolName $toolName -category $category -executablePath $executablePath -consoleApp $consoleApp -arguments $arguments
             Install-BinFile -Name $toolName -Path $executablePath
         }
-        return $executablePath
+        return ,@($toolDir, $executablePath)
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:402]: Strange syntax with this comma. Maybe we should add a comment to clarify what this does, but this shouldn't block the PR as the packages are broken.

**Review by Ana06 — APPROVED** [review]
